libvast/vast/detail/posix.hpp

```diff
@@ -18,6 +22,27 @@ namespace vast::detail {
 
 enum class socket_type { datagram, stream, fd };
 
+/// Holds the necessary state to send unix datagrams to a destination socket.
+struct uds_datagram_sender {
+  /// The destructor unlinks the src socket.
+  ~uds_datagram_sender();
```

Comment:
I struggle to understand the contract of this struct. Is it safely copyable? Is it movable? 

Can we express this contract more clearly by using a scope guard for the file descriptor?

Reply:
I changed the implementation to make it clear that it is move only.